scripts/lib/CIME/user_mod_support.py

```diff
@@ -25,6 +25,15 @@ def apply_user_mods(caseroot, user_mods_path):
             os.remove(shell_command_file)
 
     include_dirs = build_include_dirs_list(user_mods_path)
+
+    # Clean all user_nl_xxx files
```

Comment:
This functionality doesn't belong in the `apply_user_mods` function. So if we keep this functionality at all, it should be put in its own function, called separately from case.setup.